packages/addon-dev/src/rollup-public-entrypoints.ts

```diff
@@ -30,19 +28,34 @@ export default function publicEntrypoints(args: {
             id: join(args.srcDir, name),
             fileName: normalizeFileExt(name),
           });
+
         } else {
           // this file didn't match one of the user's patterns, so it must match
           // our hbsPattern. Infer the possible existence of a synthesized
           // template-only component JS file and test whether that file would
           // match the user's patterns.
           let normalizedName = normalizeFileExt(name);
           let id = join(args.srcDir, normalizedName);
-          if (
-            args.include.some((pattern) =>
-              minimatch(normalizedName, pattern)
-            ) &&
-            !pathExistsSync(id)
-          ) {
+          let normalizedMatch = args.include.some((pattern) =>
+            minimatch(normalizedName, pattern)
+          );
+          let normalizedExists = pathExistsSync(id);
+
+          let matchesDeferred = [...args.include, '**/*.ts'].some(pattern => minimatch(name, pattern) )
```

Comment:
This is what I'm least certain about :-\

Reply:
if I _only_ do: 
```ts
            this.emitFile({
              type: 'chunk',
              id: join(args.srcDir, name),
              fileName: normalizedName,
            });
```
then we get: 
```
(!) The emitted file "components/demo/index.js" overwrites a previously emitted file of the same name.
```
and the resulting component ends up being a template-only component -- ie, the class is lost

Reply:
I would guess the TS plugin is already emitting that JS file. But it probably shouldn't be. Perhaps it has options to control that?

For example, it might be trying to emit a JS file for every TS file, but that would be wrong. We only want to emit public entrypoints, and let other modules get rolled up.

Reply:
> Perhaps it has options to control that?

plugins can't interact with other plugins tho? or can they?

maybe I don't understand?

Reply:
I mean the options you're passing directly to the ts plugin in the example config.

Reply:
there are no options for passing entryPoints to the plugin? https://github.com/wessberg/rollup-plugin-ts

Reply:
@NullVoxPopuli Hi, we use rollup-plugin-ts in a recent project and found it ditches the default types of generics in the output.

For example, for input like:

```ts
export interface Foo<T extends Bar | undefined = undefined> {
  ...
}
```

the output is:

```ts
export interface Foo <T extends Bar | undefined> {
  ...
}
```

So we end up to use rollup-plugin-typescript2 + rollup-plugin-dts instead. Just for your information, because this project is not an embroider project, we can not be sure if rollup-plugin-ts will be a problem here.

Reply:
I think something else may have been at play in your project because rollup-plugin-ts very faithfully generates type definitions (I just confirmed just now by looking at the dists' d.ts files with default type args).